src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java

```diff
@@ -157,10 +166,10 @@ public void onColonyTick(@NotNull final IColony colony)
         {
             if (!colony.getRaiderManager().areSpiesEnabled())
             {
-                final int amount = InventoryUtils.getItemCountInItemHandler(this.getTileEntity().getInventory(), Items.GOLD_INGOT);
+                final int amount = InventoryUtils.getItemCountInItemHandlers(getHandlers(), i -> i.getItem().equals(Items.GOLD_INGOT));
                 if (amount >= SPIES_GOLD_COST)
                 {
-                    InventoryUtils.removeStackFromItemHandler(tileEntity.getInventory(), new ItemStack(Items.GOLD_INGOT, SPIES_GOLD_COST), SPIES_GOLD_COST);
```

Comment:
If the barracks has a list of containers, then tileEntity.getInventory should be the combined itemHandler of all them and not cause problems. If this is not the case, we maybe need the barracks tileEntity to follow the same handler code as the others. But I beleive that should not be the case.

Reply:
I felt that should be the case but ultimately when I use tileEntity.getInventory() it only checks the hut block's inventory for the gold rather than the racks. Maybe this isn't the most graceful solution but I'm not sure exactly where that problem lies otherwise. Any ideas?

Reply:
ohhh, yeah it shouldn't be getInventory, it should be getCapability with the itemhandler

Reply:
That seems to have done the trick!

Reply:
We can even make this more simple tryRemoveStackFromItemHandler(itemhandler, new ItemStack(gold, count)) 
if that returns true, set enabled.

Reply:
Also, please setup our formatter for this

Reply:
I'm now using the tryRemoveStackFromItemHandler, seems much more compact